### ISSUE TITLE
Encountered model file not found error during endpoint access

### ISSUE THREAD

**Issue by nsubram** [Hydrospheredata/hydro-serving#284]
### Runtime error
1. Model container runtime and perhaps manager runtime
2. Attached logs

### Description
We are experimenting with Hydrosphere serving and following the instructions mentioned in the Quickstart tutorial of the documentation. But encountering model not found error when trying to access the endpoint/model. Model build and the application creation worked fine as seen in screenshot. However the model.h5 is not found in the model container file system or the manager container file system. The project structure and file contents is exactly same as mentioned in the tutorial. Request you to kindly check.

![application-create-success](https://user-images.githubusercontent.com/40028371/61398404-93153e00-a8e9-11e9-812f-3c60a9cce432.png)
![application-creation](https://user-images.githubusercontent.com/40028371/61398409-93153e00-a8e9-11e9-82b7-ad3da8782927.png)
![hydrosphere-build-information](https://user-images.githubusercontent.com/40028371/61398410-93add480-a8e9-11e9-85b2-2b8e1fdbf22f.png)
![manager-log](https://user-images.githubusercontent.com/40028371/61398412-94466b00-a8e9-11e9-9dc5-125338b245af.png)
![managerui-log](https://user-images.githubusercontent.com/40028371/61398414-95779800-a8e9-11e9-9958-8a50a1f7697f.png)
![model-build-success](https://user-images.githubusercontent.com/40028371/61398416-96102e80-a8e9-11e9-82d1-b6b5ef043b0b.png)
![model-container-startup-log](https://user-images.githubusercontent.com/40028371/61398417-96a8c500-a8e9-11e9-8492-986a5af3923f.png)
![model-h5-file-not-seen-in-manager-container](https://user-images.githubusercontent.com/40028371/61398420-97d9f200-a8e9-11e9-8b5c-f715968d2697.png)
![model-h5-file-not-seen-in-model-container](https://user-images.githubusercontent.com/40028371/61398422-98728880-a8e9-11e9-9970-650477b89355.png)
![query-endpoint-failure](https://user-images.githubusercontent.com/40028371/61398424-99a3b580-a8e9-11e9-9c04-2c3439aea0ae.png)

---

**Comment by KineticCookie** [comment]
Hi.
"Payload" output of CLI shows only `src` folder and `requirements.txt`.
Could you please show the model folder structure and `payload` section in `serving.yaml` file?

**Comment by KineticCookie** [comment]
For instance in [this](https://hydrosphere.io/serving-docs/latest/tutorials/index.html) quickstart we define payload with our model file:
```
payload:
  - "src/"
  - "requirements.txt"
  - "linear_regression/model.h5"
```
I'm not sure which quickstart tutorial you use, so could you please send me the link. This way I could revise it and check for possible errors. Thanks.

**Comment by nsubram** [comment]
Hi - please find attachments for the project/folder structure and also the serving.yaml file contents.
Also am referring the following link to try out:
[https://hydrosphere.io/serving-docs/latest/tutorials/index.html](url)

![Screenshot from 2019-07-18 15-30-24](https://user-images.githubusercontent.com/40028371/61449135-f185ff00-a971-11e9-85cd-92b8894857a8.png)
![Screenshot from 2019-07-18 15-30-40](https://user-images.githubusercontent.com/40028371/61449136-f185ff00-a971-11e9-8313-85e30a241164.png)

**Comment by KineticCookie** [comment]
Yeah, I see it now, it's a typo in the docs. You have `linear_regression/model.h5` in your payload, but the actual file path is just `model.h5`. I will fix it ASAP.

However, it's very strange that CLI didn't throw an error handling this missing file.
Could you also send here the result of `hs --version` command. Thanks.

**Comment by nsubram** [comment]
Thanks for your update. I will try changing serving.yaml by removing the prefix "linear_regression" from the payload property. Also will provide the CLI output after re-trying with faulty path.

**Comment by nsubram** [comment]
Please find the faulty run screenshot: The hs tool seems to ignore the file mentioned in the payload but not found and goes ahead silently. Thanks once again for your help to spot the issue.

![Screenshot from 2019-07-17 11-21-47](https://user-images.githubusercontent.com/40028371/61462837-f908d080-a990-11e9-9b31-c002a294f0a2.png)

**Comment by nsubram** [comment]
Encountering the following issue now after correcting the payload path of the model file in the serving.yaml file. Request you to kindly check.

![Screenshot from 2019-07-18 19-23-48](https://user-images.githubusercontent.com/40028371/61464801-83066880-a994-11e9-8244-cf73464cfe6d.png)
![Screenshot from 2019-07-18 19-37-56](https://user-images.githubusercontent.com/40028371/61464802-839eff00-a994-11e9-90ad-740268f3a640.png)
![Screenshot from 2019-07-18 19-39-42](https://user-images.githubusercontent.com/40028371/61464803-839eff00-a994-11e9-9a75-2c26b9a33155.png)
![Screenshot from 2019-07-18 19-43-26](https://user-images.githubusercontent.com/40028371/61464804-839eff00-a994-11e9-8747-7cd45402a123.png)

**Comment by nsubram** [comment]
Hi
1) Did some trial and error on this issue. 
Looks like the problem is with the keras version and when I changed it as below, it works
Used Keras version 2.2.4 instead of the recommended 2.2.0
Used Tensorflow version 1.11.0 instead of the recommended 1.8.0

2) For the increment_app example in the documentation, there was a typo "requets_number"
response_number = request_number + 1

3) Further for the increment_app example in the documentation, there was another change needed to make it work (attached issue screenshot) as follows:
From:
response_tensor_shape = [hs.TensorShapeProto.Dim(size=dim) for dim in number.tensor_shape.dim]
To:
response_tensor_shape = [hs.TensorShapeProto.Dim(size=-1) for dim in number.tensor_shape.dim]


![Screenshot from 2019-07-19 08-56-55](https://user-images.githubusercontent.com/40028371/61512452-9c9cc400-aa17-11e9-8298-a6bf59b00408.png)
![Screenshot from 2019-07-19 09-09-18](https://user-images.githubusercontent.com/40028371/61512453-9d355a80-aa17-11e9-8306-62783f4958e3.png)
![Screenshot from 2019-07-19 10-38-06](https://user-images.githubusercontent.com/40028371/61512454-9d355a80-aa17-11e9-8c9a-e65e443bd08f.png)
![Screenshot from 2019-07-19 10-59-34](https://user-images.githubusercontent.com/40028371/61512455-9d355a80-aa17-11e9-86df-4127940e02dd.png)
